src/network/port.c

```diff
@@ -617,6 +617,22 @@ static int swPort_http_static_handler(swServer *serv, swHttpRequest *request, sw
     p += n;
     *p = 0;
 
+    char real_path[PATH_MAX];
+    if (!realpath(buffer.filename, real_path))
+    {
+        return SW_FALSE;
+    }
+
+    if (real_path[serv->document_root_len] != '/')
+    {
+        return SW_FALSE;
+    }
+
+    if (strncmp(real_path, serv->document_root, serv->document_root_len) != 0)
```

Comment:
if "document_root" is not a real path, the "document_root" is unavailable

Reply:
fixed in #2513 , Thanks for your review